test/api.test.js

```diff
@@ -1,169 +1,364 @@
+    'result contains line');
+  t.notEqual(selected.indexOf(pointId), -1,
+    'result contains point');
+  Draw.changeMode('simple_select', { featureIds: [polygonId] });
+  const nextSelected = Draw.getSelectedIds();
+  t.equal(nextSelected.length, 1,
+    'updates length');
+  t.equal(nextSelected[0], polygonId,
+    'updates content');
+  t.end();
+});
 
-    Draw.changeMode('draw_'+Draw.types.POLYGON);
-    var featureCollection = Draw.getAll();
-    t.equals(featureCollection.features[0].geometry.type, 'Polygon', 'Polygon');
+test('Draw.set', t => {
+  const point = getGeoJSON('point');
+  const line = getGeoJSON('line');
+  const polygon = getGeoJSON('polygon');
 
-    Draw.changeMode('draw_'+Draw.types.POINT);
-    var featureCollection = Draw.getAll();
-    t.equals(featureCollection.features[0].geometry.type, 'Point', 'Point');
+  // First set it to one collection
+  const collection = {
+    type: 'FeatureCollection',
+    features: [point, line, polygon]
+  };
+  const returnValue = Draw.set(collection);
+  t.equal(returnValue.length, 3,
+    'return value is correct length');
+  const pointId = returnValue[0];
+  const lineId = returnValue[1];
+  const polygonId = returnValue[2];
+  t.equal(Draw.get(pointId).geometry.type, 'Point',
+    'point id returned');
+  t.equal(Draw.get(lineId).geometry.type, 'LineString',
+    'line id returned');
+  t.equal(Draw.get(polygonId).geometry.type, 'Polygon',
+    'polygon id returned');
+  t.equal(Draw.getAll().features.length, 3,
+    'all features loaded');
 
-    Draw.changeMode('draw_'+Draw.types.LINE);
-    var featureCollection = Draw.getAll();
-    t.equals(featureCollection.features[0].geometry.type, 'LineString', 'Line');
-    Draw.deleteAll();
+  // Then set to another
+  addSpy.reset();
+  deleteSpy.reset();
+  const nextCollection = {
+    type: 'FeatureCollection',
+    features: [polygon]
+  };
+  const nextReturnValue = Draw.set(nextCollection);
+  t.equal(nextReturnValue.length, 1,
+    'return value is correct length');
+  const nextPolygonId = nextReturnValue[0];
+  t.equal(Draw.get(nextPolygonId).geometry.type, 'Polygon',
+    'polygon id returned');
+  t.equal(Draw.getAll().features.length, 1,
+    'all features replaced with new ones');
+  t.ok(addSpy.calledWith(nextCollection),
+    'Draw.add called with new collection');
+  t.equal(deleteSpy.callCount, 1,
+    'Draw.delete called');
+  t.deepEqual(deleteSpy.getCall(0).args, [[
+    pointId,
+    lineId,
+    polygonId
+  ]], 'Draw.delete called with old features');
 
-    t.end();
-  });
+  // Then set to another that contains a feature
+  // with an already-used id
+  addSpy.reset();
+  deleteSpy.reset();
+  const newLine = getGeoJSON('line');
+  const overlappingPolygon = getGeoJSON('polygon');
+  overlappingPolygon.id = nextPolygonId;
+  const overlappingCollection = {
+    type: 'FeatureCollection',
+    features: [newLine, overlappingPolygon]
+  };
+  const overlappingReturnValue = Draw.set(overlappingCollection);
```

Comment:
I guess I was just being generic because we're testing the return value. I can change all "returnValue" to "drawInstance", so it matches the expected result.